src/System.ServiceProcess.ServiceController/tests/ServiceBaseTests.cs

```diff
@@ -164,48 +164,29 @@ public void TestOnContinueBeforePause()
         [ConditionalFact(nameof(IsElevatedAndSupportsEventLogs))]
         public void LogWritten()
         {
-            using (EventLog eventLog = new EventLog("Application"))
-            {
-                ServiceBase sb = new ServiceBase() { ServiceName = nameof(LogWritten) + Guid.NewGuid().ToString() };
-                Assert.False(EventLog.SourceExists(sb.ServiceName));
-                try
-                {
-                    ServiceBase.Run(sb);
-                    eventLog.Source = sb.ServiceName;
-                    Assert.True(EventLog.SourceExists(sb.ServiceName));
-                }
-                finally
-                {
-                    sb.Stop();
-                    EventLog.DeleteEventSource(sb.ServiceName);
-                }
-            }
+            string serviceName = Guid.NewGuid().ToString();
+            // The default username for installing the service is NT AUTHORITY\\LocalService which does not have access to EventLog.
+            // If the username is null, then the service is created under LocalSystem Account which have access to EventLog.
+            var _testService = new TestServiceProvider(serviceName, userName: null);
+            Assert.True(EventLog.SourceExists(serviceName));
+            _testService.DeleteTestServices();
         }
 
         [ConditionalFact(nameof(IsElevatedAndSupportsEventLogs))]
         public void LogWritten_AutoLog_False()
         {
```

Comment:
Why aren't we setting AutoLog to false anymore?  If it's because it's the default, we should validate that with an assert.

Reply:
we are setting it at https://github.com/dotnet/corefx/pull/28939/files#diff-933e606564235cff6edb82696a9e8a1bR36

Reply:
We cant validate the assert directly as there test service is started in another process and we cant have testService Object in the test. So we are testing it through the source property of eventLog